components/provisioner/assets/database/provisioner.sql

```diff
@@ -0,0 +1,91 @@
+-- Cluster
+
+CREATE TABLE cluster
+(
+    ID uuid PRIMARY KEY CHECK (id <> '00000000-0000-0000-0000-000000000000'),
+    kubeconfig text,
+    terraform_state json,
+    creation_timestamp timestamp without time zone NOT NULL
+);
+
+
+-- Cluster Config
+
+CREATE TYPE infrastructure_provider AS ENUM (
+    'GARDENER',
+    'GCP',
+    'AKS'
+    );
+
+CREATE TABLE cluster_config
+(
+    ID uuid PRIMARY KEY CHECK (id <> '00000000-0000-0000-0000-000000000000'),
+    name varchar(256) NOT NULL,
+    node_count integer NOT NULL,
+    disk_size varchar(256) NOT NULL,
+    machine_type varchar(256) NOT NULL,
+    compute_zone varchar(256) NOT NULL,
+    version varchar(256) NOT NULL,
+    infrastructure_provider infrastructure_provider NOT NULL,
+    runtime_id uuid NOT NULL,
```

Comment:
I'm not sure if runtime_id is a suitable name here. The field foreign key is built on refers to cluster table so maybe cluster_id is better.